monitor.py

```diff
@@ -158,17 +159,30 @@ def run(self):
         if start == self.NEVER:
             start = None
 
-        cutoff = datetime.datetime.utcnow()
-        new_timestamp_value = self.run_once(start, cutoff) or cutoff
-        duration = datetime.datetime.utcnow() - cutoff
-        self.cleanup()
+        exception = None
+        started_running = datetime.datetime.utcnow()
+        try:
+            new_timestamp_value = (
+                self.run_once(start, started_running) or started_running
+            )
```

Comment:
This is a little annoying; I wrote code assuming that `timestamp` was the end time of the previous run and `start` was the start time. But for Monitors, the `timestamp` is the _start_ time of the previous run.

I'm not sure what to do about this; maybe I should change `Monitor` to look for its time in `Timestamp.start` and use `Timestamp.timestamp` just for display purposes?

Reply:
Yeah, I would change Monitor to use Timestamp.start to be consistent with the other scripts.